packages/core/src/reducers/activities.js

```diff
@@ -97,7 +81,8 @@ export default function activities(state = DEFAULT_STATE, { meta, payload, type
 
     case INCOMING_ACTIVITY:
       // UpdateActivity is not supported right now because we ignore duplicated activity ID
-      if (!~state.findIndex(({ id }) => id === payload.activity.id)) {
+      // TODO: [P4] Move "typing" into Constants.ActivityType
```

Comment:
Agree, we should be able to use typing from `botframework-directlinejs` without loading the library.